js/ui/scheduler/header/header.js

```diff
@@ -183,24 +185,32 @@ export class SchedulerHeader extends Widget {
     }
 
     _updateCurrentDate(date) {
```

Comment:
Let's rename it and leave the code inside of it as is:

```typescript
  _handleCurrentDateChange(date) {
    const onCurrentDateChange = this.option('onCurrentDateChange');
    onCurrentDateChange(date);

    this._callEvent('currentDate', date);
  }
```

And add the new method with the same name ```_updateCurrentDate```:
```typescript
_updateCurrentDate(date) {
  this._handleCurrentDateChange(date);
  this._calendar.option('value', date);
}
```

If we do it, we should use the ```__handleCurrentDateChange``` [here](https://github.com/DevExpress/DevExtreme/pull/24914/files#diff-841e3cf44beb9f50697a667bde93fa69432c1d828f2642c500e64b774dc43927R213) and leave second usage of the  ```_updateCurrentDate``` without changes :)